datacube/utils/geometry/_base.py

```diff
@@ -972,66 +978,61 @@ def __eq__(self, other):
                 and self.crs == other.crs)
 
 
-def relative_translation(A: GeoBox, B: GeoBox):  # pylint: disable=invalid-name
+def bounding_box_in_pixel_domain(reference: GeoBox, other: GeoBox):
```

Comment:
Not sure about the order of these two parameters.

I think calling this function this way would be cleaner:

```python
bounding_box_in_pixels(some_geobox, reference=reference_geobox)
```

Reply:
then you won't need to call it `other` which is a poor name